src/borg/item.pyx

```diff
@@ -393,16 +400,27 @@ class ManifestItem(PropDict):
 
     If a ManifestItem shall be serialized, give as_dict() method output to msgpack packer.
     """
+    VALID_KEYS = {'version', 'archives', 'timestamp', 'config', 'item_keys', }
 
-    VALID_KEYS = {'version', 'archives', 'timestamp', 'config', 'item_keys', }  # str-typed keys
+    version = PropDictProperty(int)
+    archives = PropDictProperty(dict)  # name -> dict
+    timestamp = PropDictProperty(str, 'surrogate-escaped str', encode=safe_encode, decode=safe_decode)
+    config = PropDictProperty(dict)
+    item_keys = PropDictProperty(tuple)
 
-    __slots__ = ("_dict", )  # avoid setting attributes not supported by properties
 
-    version = PropDict._make_property('version', int)
-    archives = PropDict._make_property('archives', dict)  # name -> dict
-    timestamp = PropDict._make_property('timestamp', str, 'surrogate-escaped str', encode=safe_encode, decode=safe_decode)
-    config = PropDict._make_property('config', dict)
-    item_keys = PropDict._make_property('item_keys', tuple)
+cpdef _init_names():
+    """
+    re-implements python __set_name__
+    """
+    cdef PropDictProperty val = None
+    for cls in PropDict.__subclasses__():
+         for name, value  in vars(cls).items():
+             if isinstance(value, PropDictProperty):
+                 val = value
```

Comment:
what's the point of this?

Reply:
Access to c levels attributes, but I'm thinking about having readonly public there